modules/clightning/module.cpp

```diff
@@ -65,20 +71,60 @@ std::string clightning_des_invoice(const std::string& input) {
         result << invoice->description;
     }
 
+    result << ";METADATA=";
+    if (invoice->metadata) {
+        result << hex_encode(invoice->metadata, tal_bytelen(invoice->metadata));
+    }
+
     struct pubkey key;
     assert(pubkey_from_node_id(&key, &invoice->receiver_id));
 
     uint8_t compressed[33];
     pubkey_to_der(compressed, &key);
     result << ";RECIPIENT=" << hex_encode(compressed, 33);
 
+    result << ";DESCRIPTION_HASH=";
+    if (invoice->description_hash) {
+        result << hex_encode(invoice->description_hash->u.u8, 32);
+    }
+
     // Convert the expiry time to seconds, ensuring consistent overflow behavior
     // with other implementations like LND. The multiplication and division by 1000000000
     // may appear redundant, but it's intentionally mirroring the nanosecond conversion 
     // logic used in LND to ensure the same overflow characteristics.
     result << ";EXPIRY=" << (invoice->expiry * 1000000000) / 1000000000;
+
+    result << ";MIN_FINAL_CLTV_EXPIRY_DELTA=" << invoice->min_final_cltv_expiry;
     result << ";TIMESTAMP=" << invoice->timestamp;
-    result << ";ROUTING_HINTS=" << tal_count(invoice->routes);
+
+
+    result << ";FALLBACK_ADDRESS=";
+    if (invoice->fallbacks) {
+        std::string addr = encode_scriptpubkey_to_addr(tmpctx, params, invoice->fallbacks[0]);
+        result << addr;
+    }
+
+    result << ";PRIVATE_ROUTES=";
+    if (invoice->routes) {
+        for (size_t i = 0; i < tal_count(invoice->routes); i++) {
+            struct route_info *route = invoice->routes[i];
+            result << ";ROUTE " << i << "=";
+
+            for (size_t j = 0; j < tal_count(route); j++) {
+                result << ";HOP " << j << "=";
+                struct pubkey key;
+                assert(pubkey_from_node_id(&key, &route[j].pubkey));
+                uint8_t compressed[33];
+                pubkey_to_der(compressed, &key);
+                result << ";NODE_ID=" << hex_encode(compressed, 33);
+                result << ";SHORT_CHANNEL_ID=" << route[j].short_channel_id.u64;
+                result << ";FEES=" << route[j].fee_base_msat << ";" << route[j].fee_base_msat;
+                result << ";CLTV_EXPIRY_DELTA=" << route[j].cltv_expiry_delta;
+                result << ";PROPORTIONAL_MILLIONTHS=" << route[j].fee_proportional_millionths;
+            }
+        }
+    }
```

Comment:
This is going to create a confusing string:  `PRIVATE_ROUTES=;ROUTE 0=;HOP 0=;NODE_ID=...;HOP 1=;NODE_ID=...;ROUTE 1=...`

Can we do a more readable format? Something like:

```
PRIVATE_ROUTE=(NODE_ID=...,SHORT_CHANNEL_ID=...,FEES=...,CLTV_EXPIRY_DELTA=...,PROPORTIONAL_MILLIONTHS=...),(NODE_ID=...,SHORT_CHANNEL_ID=...,FEES=...,CLTV_EXPIRY_DELTA=...,PROPORTIONAL_MILLIONTHS=...);PRIVATE_ROUTE=(NODE_ID=...,SHORT_CHANNEL_ID=...,FEES=...,CLTV_EXPIRY_DELTA=...,PROPORTIONAL_MILLIONTHS=...),...;
```